CASSANDRA-17092 CEP-15 Accord Beta

maedhroz opened the pull request.

maedhroz committed:
CEP-15: Multi-Partition Transaction CQL Support (Alpha)

patch by Caleb Rackliffe; reviewed by David Capwell for CASSANDRA-17719

Co-authored-by: Caleb Rackliffe <calebrackliffe@gmail.com>
Co-authored-by: Blake Eggleston <bdeggleston@gmail.com>
Co-authored-by: David Capwell <dcapwell@apache.org>

belliottsmith committed:
Partial Replication of Transaction State

The replication of transactions and their processing state must be optimal, i.e. linearly proportional to the size of the transaction and the number of keys involved. This patch ensures a transaction is replicated in slices, with each slice being written (replication factor) times only to the shard that owns the keys, and only once (except in the case the transaction is recovered).

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-17109

bdeggleston committed:
CEP-15 (C*): Messaging and storage engine integration

patch by Blake Eggleston; reviewed by Benedict Elliott Smith, David Capwell for CASSANDRA-17103

belliottsmith committed:
CEP-15: Routables
  - Integrate accord-core changes for CASSANDRA-18087

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-18087

dcapwell committed:
Command table now uses a local version added to the payload rather than rely on a version column, and fixed a few feedback related issues

patch by David Capwell; reviewed by Blake Eggleston for CASSANDRA-17103

belliottsmith committed:
Transaction Expiration

Ensure old or abandoned transactions are invalidated wherever possible.

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-18041

belliottsmith committed:
Invalidation fixes/improvements
 - Integrate accord-core changes for CASSANDRA-18057

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-18057

belliottsmith committed:
Refactor response tracking to improve efficiency and clarity; introduce dedicated property tests; re-activate fast-path during range movements

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-18056

aweisberg committed:
Support CAS and serial read on Accord

patch by Ariel Weisberg; Reviewed by Caleb Rackliffe for CASSANDRA-18100

belliottsmith committed:
Shard local CommandStores on contiguous ranges

patch by Benedict; reviewed by Aleksey Yeschenko for CASSANDRA-18142

belliottsmith committed:
Refactor Timestamp/TxnId
 - Combine real and logical into a single 64-but HLC
 - Introduce 16 flag bits
 - Pack epoch (48-bits), HLC (64-bits) and flags (16-bits) into two longs in memory

patch by Benedict; reviewed by Aleksey Yeschenko for CASSANDRA-18172

belliottsmith committed:
Introduce RangeDeps

Refactor Deps into KeyDeps and RangeDeps

patch by Benedict; reviewed by Aleksey Yeschenko for CASSANDRA-18173

belliottsmith committed:
Introduce Range transactions

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-18174

belliottsmith committed:
ninja: pick a cassandra-accord SHA to build against to decouple source trees

dcapwell committed:
CASSANDRA-18154: CEP-15: Enhance returning SELECT to allow partition and clustering IN clauses to return multiple partitions/rows

dcapwell committed:
CEP-15: (C*) Enhance returning SELECT to allow partition and clustering IN clauses to return multiple partitions/rows

patch by David Capwell; reviewed by Caleb Rackliffe for CASSANDRA-18154

aweisberg committed:
Accord support for live migration

Patch by Ariel Weisberg; Reviewed by Blake Eggleston for CASSANDRA-18129

maedhroz committed:
Post-rebase fixes around CASSANDRA-18099 and CASSANDRA-18114 (among other things)

bdeggleston committed:
add AsyncChain implementations and tests

Patch by Blake Eggleston; Reviewed by David Capwell & Benedict Elliott Smith for CASSANDRA-18004

maedhroz committed:
Feature Flag for Accord Transactions

patch by Caleb Rackliffe; reviewed by David Capwell for CASSANDRA-18195

iamaleksey committed:
Switch Node.Id from long to int

patch by Aleksey Yeschenko; reviewed by Benedict Elliott Smith for
CASSANDRA-18135

dcapwell committed:
Ninja: CASSANDRA-18214 added rat checks to Accord, but our metadata files .BRANCH and .REPO do not have a license, causing the build to fail; exclude rat when building accord

patch by David Capwell; reviewed by Ariel Weisberg for CASSANDRA-18214

maedhroz committed:
use equals() rather than reference equality for excluding ourselves from the dependency builder

patch by Benedict; reviewed by Caleb Rackliffe for CASSANDRA-18218

bdeggleston committed:
CEP-15/Accord Immutable State Refactor

Patch by Blake Eggleston; Reviewed by David Capwell and Benedict Elliott Smith for Cassandra-18192

jacek-lewandowski committed:
Fix null value handling for static columns

patch by <jacek-lewandowski>; reviewed by <maedhroz> and <dcapwell> for CASSANDRA-18241

iamaleksey committed:
CEP-15: Store PreAccept, Accept, Commit, and Apply messages
in a durable log before processing by CommandStores

patch by Aleksey Yeschenko; reviewed by David Capwell for
CASSANDRA-18344

jacek-lewandowski committed:
Fix statement validation against partition range queries

patch by Jacek Lewandowski; reviewed by Caleb Rackliffe for CASSANDRA-18240

dcapwell committed:
CEP-15: (C*) Add git submodule for Accord

patch by David Capwell; reviewed by Caleb Rackliffe, Michael Semb Wever for CASSANDRA-18204

dcapwell committed:
Switched away from https endpoints for git submodules in favor of relative paths during development

patch by David Capwell; reviewed by Caleb Rackliffe, Michael Semb Wever for CASSANDRA-18204

dcapwell committed:
Add support for prepared statements for accord transactions

patch by David Capwell; reviewed by Alex Petrov, Caleb Rackliffe, Jacek Lewandowski for CASSANDRA-18299

dcapwell committed:
Remove git hook for pre-push as it is redundant and causes issues when merging to mainline

patch by David Capwell; reviewed by Caleb Rackliffe for CASSANDRA-18309

dcapwell committed:
CEP-15: (Accord) Migrate Accord away from JDK random to a new interface RandomSource

patch by David Capwell; reviewed by Blake Eggleston for CASSANDRA-18213

jacek-lewandowski committed:
Improve transaction statement validation

patch by Jacek Lewandowski; reviewed by David Capwell and Caleb Rackliffe for CASSANDRA-18302

maedhroz committed:
fixing remaining (mostly compilation) issues after initial rebase of cep-15-accord on cep-21-tcm

dcapwell committed:
Operations.migrateReadRequiredOperations fails due to concurrent access when TransactionStatement is prepared

patch by David Capwell; reviewed by Ariel Weisberg, Caleb Rackliffe for CASSANDRA-18337

dcapwell committed:
CEP-15 (Accord) Expected reply message with verb ACCORD_INFORM_OF_TXNID_RSP but got ACCORD_SIMPLE_RSP

patch by David Capwell; reviewed by Caleb Rackliffe for CASSANDRA-18375

dcapwell committed:
CEP-15 (Accord) Original and recover coordinators may hit a race condition with PreApply where reads and writes are interleaved, causing one of the coordinators to see the writes from the other

patch by David Capwell; reviewed by Ariel Weisberg for CASSANDRA-18422

belliottsmith committed:
(Accord only) Permit nodes to join a cluster without the full transaction history

patch by Benedict; reviewed by Blake Eggleston for CASSANDRA-18523

dcapwell committed:
CEP-15: (C*) Accord message processing should avoid being passed on to a Stage and run directly in the messageing handler

patch by David Capwell; reviewed by Ariel Weisberg, Benedict Elliott Smith for CASSANDRA-18364

aweisberg committed:
Accord/non-Accord interoperability

Patch by Ariel Weisberg and Blake Eggleston; Reviewed by Blake Eggleston for CASSANDRA-18129

Co-authored-by: Blake Eggleston <beggleston@apple.com>

dcapwell committed:
CEP-15: (C*) Enhance in-memory FileSystem to work with mmap and support tests to add custom logic

patch by David Capwell; reviewed by Caleb Rackliffe for CASSANDRA-18485

dcapwell committed:
CEP-15: (C*) Add notion of CommandsForRanges and make this durable in C*

patch by David Capwell; reviewed by Ariel Weisberg for CASSANDRA-18519

Commit:
minor cleanup in Verb and BTreeSet

Commit:
removing unnecessary calls to ServerTestUtils.daemonInitialization() in a handful of tests

Commit:
make sure workspace.xml specifies a storagedir

iamaleksey committed:
CEP-15: Extend Accord MessageType with a side effect flag

patch by Aleksey Yeschenko; reviewed by Benedic Elliott Smith for
CASSANDRA-18561

iamaleksey committed:
CEP-15: Convert AccordStateCache cache from write-through to write-back

patch by Aleksey Yeschenko; reviewed by Blake Eggleston for
CASSANDRA-18563

bdeggleston committed:
CEP-15: Accord Bootstrap Integration

Patch by Blake Eggleston and Benedict Elliott Smith; Reviewed by David
Capwell for CASSANDRA-17101

CEP-15: Accord TCM integration

Patch by Blake Eggleston; Reviewed by David Capwell for CASSANDRA-18444

iamaleksey committed:
CEP-15: Minimize transaction state kept in system tables

patch by Aleksey Yeschenko; reviewed by Ariel Weisberg for CASSANDRA-18573

jonmeredith committed:
Several simulator fixes not yet merged to cep-15-accord

s c4bf1533b4 Add more JVM arg checks to the paxos simulation runner
s 1b650a0636 break circular dependency between FileSystemOwnershipCheck and CassandraRelevantProperties
s e0111748dd fix verify with-rng and interrupts
s f8475f2a18 make SimulatedFailureDetector deterministic
s ee77734bbe Deterministic transformation
s 890f23696a Add mechanism forsimulator to intercept password salting
s 5205b99874 Specify Paxos V2 because it is the only linearizable option

Patch by Ariel Weisberg; Reviewed by David Capwell for CASSANDRA-19008

belliottsmith committed:
Support state eviction (i.e. permit the state machine to erase transactions that are known to be applied across the cluster)

patch by Benedict Elliott Smith; reviewed by Ariel Weisberg, Aleksey Yeschenko, and David Capwell for CASSANDRA-18883

Co-authored-by: Benedict Elliott Smith <benedict@apache.org>
Co-authored-by: Ariel Weisberg <aweisberg@apple.com>
Co-authored-by: Aleksey Yeschenko <aleksey@apache.org>
Co-authored-by: David Capwell <dcapwell@gmail.com>

dcapwell committed:
CEP-15 (C*) When a host replacement happens don't loose the peer mapping right away (#3575)


patch by David Capwell; reviewed by Blake Eggleston for CASSANDRA-18764

dcapwell committed:
Fix Accord compaction purger tombstone logic

Accord compaction purgers see random slices of Accord state during compaction (based on randomly selected compaction inputs).

For at least the `durability` column in the `commands` table the tombstone being created when truncating was deleting the latest value since we can get enough information to truncate without actuall yhaving the latest `durability` value.

To fix we can wait to emit a tombstone until we are erasing the entire command row when truncating or truncating with outcome and meanwhile we can drop the extra columns that are no longer needed instead of using a tombstone. We don't need to emit cell tombstones we can drop them from the purger when processing each row.

patch by Ariel Weisberg; reviewed by David Capwell for CASSANDRA-18795

jacek-lewandowski committed:
CASSANDRA-18774: Fix pre-commit hook

aweisberg committed:
Allow exceptions to be propagated remotely

https://github.com/apache/cassandra-accord/pull/56

Patch by Ariel Weisberg; Reviewed by David Capwell for CASSANDRA-18779

dcapwell committed:
CEP-15 (C*): when loading commands that have empty waiting_on, make sure not to loose the partial deps (#3590)

patch by David Capwell; reviewed by Aleksey Yeschenko for CASSANDRA-18783

iamaleksey committed:
Fix CompactionAccordIteratorsTest

bdeggleston committed:
Bootstrap/TCM integration test fixes

dcapwell committed:
CEP-15: (C*) NPE on restart as epoch isnt known but verb handler doesnt delay mapping to Node.Id

dcapwell committed:
CEP-15: (C*) accord.messages.BeginRecovery.RecoverNack#supersededBy is nullable but C* serializer doesn't expect null

Commit:
- Avoid persisting fragments that do not require completion as Updates in TxnWrite, as they can simply be pulled from PartialTxn when needed in Write#apply()
- Avoid serializing full TxnData instances to Accord state tables

patch by Caleb Rackliffe; reviewed by David Capwell, Benedict Elliot Smith, and Ariel Weisberg for CASSANDRA-18355

maedhroz committed:
ninja-fix: checkstyle fixes in VirtualTableTest, HistoryValidatorTest, SeedDefiner, RunStartDefiner, and Config

iamaleksey committed:
Fix Propagate serializer

maedhroz committed:
ninja-fix: minor post-TCM-rebase cleanup

maedhroz committed:
ninja-fix: exclude accord_demo.txt from rat checks

maedhroz committed:
Fix AccordJournalTest and ConfigCompatibilityTest, and revert the sendMessageWithFailureCB() callback to printing just the repair failure reason in its error message

dcapwell committed:
Get simulator working (again)

Co-authored-by: Ariel Weisberg <aweisberg@apple.com>
Co-authored-by: Benedict Elliott Smith <benedict@apache.org>

Maxwell-Guo commented on accord_demo.txt:
what about rename this to general_purpose_transactions_demo.txt ?

maedhroz replied:
It's very likely that, before `cep-15-accord` merges, we'll remove this document entirely and replace it w/ something like the Markdown readme we have for SAI. In particular, the unsafe epoch creation is only a crutch while TCM/CEP-21 integration is still pending.

Maxwell-Guo commented on conf/cassandra.yaml:
```diff
@@ -1818,6 +1818,9 @@ drop_compact_storage_enabled: false
 # Whether or not USE <keyspace> is allowed. This is enabled by default to avoid failure on upgrade.
 #use_statements_enabled: true
 
+# Enables the execution of Accord (multi-key) transactions on this node.
```
what about change to "Enables the execution of General Purpose Transactions (multi-partition) transactions on this node." ,Together with the "accord_transactions_enabled" configuration .

maedhroz replied:
We can certainly revisit this before `cep-15-accord` merges.

maedhroz commented:
Last known CircleCI run: https://app.circleci.com/pipelines/github/maedhroz/cassandra/774/workflows/ca42a1cd-8d3b-4309-b25a-997648a28730

amitdpawar committed:
Enable Direct-IO feature for CommitLog files using Java native API's.

Patch by Amit Pawar and Jacek Lewandowski; reviewed by Branimir Lambov and Maxwell Guo for CASSANDRA-18464

Co-authored-by: Amit Pawar <Amit.Pawar@amd.com>
Co-authored-by: Jacek Lewandowski <lewandowski.jacek@gmail.com>

jacek-lewandowski committed:
CEP-15: Add Accord configuration stub

Patch by Jacek Lewandowski; reviewed by David Capwell for CASSANDRA-18221

belliottsmith committed:
Improve validation and address various discovered faults

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-19045

jacek-lewandowski committed:
CEP-15: Accord metrics

Patch by Jacek Lewandowski, reviewed by Caleb Rackliffe, David Capwell and Henrik Ingo for CASSANDRA-18580

dcapwell committed:
CEP-15: (C*) Implement TopologySorter to prioritise hosts based on DynamicSnitch and/or topology layout

patch by Blake Eggleston, David Capwell; reviewed by Blake Eggleston for CASSANDRA-18929

ifesdjeen committed:
Use pinned Harry version

bdeggleston committed:
Reduce command deps

Patch by Blake Eggleston; Reviewed by Benedict Elliott Smith for CASSANDRA-18784

szymon-miezal committed:
Fix nodetool repair_admin summarize-pending command to not throw exception

Fixed a bug causing the `OpenDataException` being thrown
when executing the `repair_admin summarize-pending` command.
This patch addresses the problem by including a missing composite
in `PendingStats.toComposite`, ensuring proper data conversion.

patch by Szymon Miezal; reviewed by Stefan Miklosovic and Brandon Williams for CASSANDRA-19014

mike-tr-adamson committed:
Improve code model around IndexContext

 - Replace IndexContext with IndexTermType and IndexDefinition
 - Move index specific managers, factories and metrics to StorageAttachedIndex
 - Refactor Expression to explicitly define indexed and unindexed expressions

 patch by Mike Adamson; reviewed by Andres de la Peña, Caleb Rackliffe for CASSANDRA-18166

aweisberg committed:
Ninja fix SlowMessageFuzzTest and InvertedIndexSearcherTest

Patch by Ariel Weisberg; Reviewed by David Capwell for CASANDRA-19023

aweisberg committed:
Fix Paxos V2 prepare response serialization

Patch by Ariel Weisberg; Reviewed by David Capwell for CASSANDRA-19023

iamaleksey committed:
Pre-requisite changes for CASSANDRA-18888

patch by Aleksey Yeschenko; reviewed by Ariel Weisberg for
CASSANDRA-18888

iamaleksey committed:
Quick fix for AccordCommandStoreTest#commandLoadSave()

zaaath committed:
Make nodetool tablestats use number of significant digits for time and average values consistently

patch by Leo Toff; reviewed by Stefan Miklosovic and Brandon Williams for CASSANDRA-19015

jacek-lewandowski committed:
Fix filtering system ks sstables for relocation on startup

Patch by Jacek Lewandowski; reviewed by Benjamin Lerer for CASSANDRA-18963

beobal committed:
Implementation of Transactional Cluster Metadata as described in CEP-21

An overview of the core components can be found in the included
TransactionalClusterMetadata.md

patch by Alex Petrov, Marcus Eriksson and Sam Tunnicliffe; reviewed by
Alex Petrov, Marcus Eriksson and Sam Tunnicliffe for CASSANDRA-18330

Co-authored-by: Marcus Eriksson <marcuse@apache.org>
Co-authored-by: Alex Petrov <oleksandr.petrov@gmail.com>
Co-authored-by: Sam Tunnicliffe <samt@apache.org>

ifesdjeen committed:
Fix rejectSubsequentInProgressSequence test: cap message sizes for commit failures

Patch by Alex Petrov; reviewed by Sam Tunnicliffe for CASSANDRA-19076.

krummas committed:
Revert bad dtest repo in circleci conf

mike-tr-adamson committed:
Fix broken indexing tests when using SAI
 - This fixes a number of broken tests when the default index is set to SAI
 - Composite partition indexes were being filtered prior to row filtering in the
   index searcher resulting in incorrect results
 - Static and non-static index intersection was failing because static primary keys
   were not comparing correctly against non-static primary keys

 patch by Mike Adamson; reviewed by Andres de la Peña, Michael Semb Wever for CASSANDRA-19034

ifesdjeen committed:
Introduce 5.1 messaging format that brings in Epoch 

Patch by Alex Petrov; reviewed by Sam Tunnicliffe CASSANDRA-19066.

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Improve code model around IndexContext

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Fix broken indexing tests when using SAI  - This fixes a number of broken tests when the default index is set to SAI  - Composite partition indexes were being filtered prior to row filtering in the    index searcher resulting in incorrect results  - Static and non-static index intersection was failing because static primary keys    were not comparing correctly against non-static primary keys

michaelsembwever committed:
Prepare debian changelog for 5.0-beta1

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Prepare debian changelog for 5.0-beta1

krummas committed:
Test failure: org.apache.cassandra.db.RepairedDataInfoTest

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19059

beobal committed:
[CEP-21] Don't log unknown endpoint errors unnecessarily

Patch by Sam Tunnicliffe; reviewed by Marcus Eriksson for CASSANDRA-19092

krummas committed:
Test failure: org.apache.cassandra.tools.JMXCompatabilityTest

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19060

krummas committed:
Test failure: org.apache.cassandra.db.CorruptPrimaryIndexTest.bigPrimaryIndexDoesNotDetectDiskCorruption

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19061

mike-tr-adamson committed:
Provide truncate task for SAI

If a table is truncated during the initial build of an index there is a chance that the index build will get interrupted and it won't get marked queryable. This patch provides a truncate task for SAI that marks the index queryable during truncation.

 patch by Mike Adamson; reviewed by Caleb Rackliffe, Michael Semb Wever for CASSANDRA-19032

krummas committed:
Test Failure: org.apache.cassandra.db.compaction.writers.CompactionAwareWriterTest.test*CompactionWriter-trie

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19091

dcapwell committed:
Ninja for CASSANDRA-19045: make sure to use https rather than git@ for submodule so its portable

dcapwell committed:
Ninja for CASSANDRA-19045: use the latest sha from trunk rather than an old one from 10 months ago

yifan-c committed:
Support max SSTable size in sorted CQLSSTableWriter

patch by Yifan Cai; reviewed by Alex Petrov, Francisco Guerrero, Maxwell Guo for CASSANDRA-18941

krummas committed:
Test Failure: org.apache.cassandra.distributed.upgrade.ClusterMetadataUpgradeTest.upgradeWithHintsTest

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19063

mike-tr-adamson committed:
Fix SAI intersection queries

 - Fix comparison in PostingListRangeIterator for updating skip token
 - Fix binary search in KeyLookup.clusteredSeekToKey
 - Added new on-disk component for storing partition sizes by partition ID

 patch by Mike Adamson; reviewed by Caleb Rackliffe, Mick Semb Wever for CASSANDRA-19011

krummas committed:
Test failure: org.apache.cassandra.distributed.test.ring.DecommissionTest.test*Version*

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19078

jacek-lewandowski committed:
Merge branch 'cassandra-4.0' into cassandra-4.1

* cassandra-4.0:
  Fix filtering system ks sstables for relocation on startup

beobal committed:
[CEP-21] Add log message for backwards compatibility

Patch by Sam Tunnicliffe; reviewed by Marcus Eriksson for CASSANDRA-19068

jacek-lewandowski committed:
Merge branch 'cassandra-4.1' into cassandra-5.0

* cassandra-4.1:
  Fix filtering system ks sstables for relocation on startup

jacek-lewandowski committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Fix filtering system ks sstables for relocation on startup

rhuffy committed:
Upgrade jackson to 2.15.3 and snakeyaml to 2.1

Patch by Raymond Huffman; reviewed by brandonwilliams and smiklosovic
for CASSANDRA-18875

jacek-lewandowski committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Enable Direct-IO feature for CommitLog files using Java native API's.

smiklosovic committed:
Do not set RPC_READY to false on transports shutdown in order to not fail counter updates for deployments with coordinator and storage nodes with transports turned off

This is the follow-up commit of CASSANDRA-18935 where we set RPC_READY to false
when transports were shut down in runtime. The problem is that the current logic in StorageProxy.findSuitableEndpoint method,
used for the selection of a leader for counter mutations, is filtering out all endpoints which do not have RPC_READY
set to true. Hence, if there is a deployment of a coordinator node (not joining a ring) and storage nodes which
have transports turned off (e.g. for security reasons), then a coordinator node will select no endpoint as a
counter mutation leader which renders counter mutations impossible.

This change just reverts the original fix which was setting RPC_READY to false when transports were shut down in runtime
(e.g. by nodetool disablebinary).

For trunk (to be 5.1 at time of writing this), there is already TCM in place and the correct fix yet to be implemented
is to decouple from checking RCP_READY state and base it e.g. on JOINED state from TCM perspective. Please follow
CASSANDRA-19103 where this will be addressed.

patch by Stefan Miklosovic; reviewed by Brandon Williams for CASSANDRA-18935

smiklosovic committed:
Merge branch 'cassandra-3.0' into cassandra-3.11

smiklosovic committed:
Merge branch 'cassandra-3.11' into cassandra-4.0

smiklosovic committed:
Merge branch 'cassandra-4.0' into cassandra-4.1

smiklosovic committed:
Merge branch 'cassandra-4.1' into cassandra-5.0

smiklosovic committed:
Merge branch 'cassandra-5.0' into trunk

krummas committed:
Clone EndpointState before sending GossipShutdown message

Patch by marcuse; reviewed by David Capwell for CASSANDRA-19115

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Provide truncate task for SAI

dmi0 committed:
Fix cassandra-stress in simplenative mode with prepared statements

patch by Dmitry Bychkov; reviewed by Stefan Miklosovic and Brandon Williams for CASSANDRA-18744

smiklosovic committed:
Merge branch 'cassandra-4.0' into cassandra-4.1

smiklosovic committed:
Merge branch 'cassandra-4.1' into cassandra-5.0

smiklosovic committed:
Merge branch 'cassandra-5.0' into trunk

smiklosovic committed:
Merge branch 'cassandra-4.0' into cassandra-4.1

smiklosovic committed:
Merge branch 'cassandra-4.1' into cassandra-5.0

smiklosovic committed:
Merge branch 'cassandra-5.0' into trunk

aweisberg committed:
Fix additional live migration/interop merge issues

s fc83325fa9 Fix AccordObjectSizes empty sizes
s 44bce6a08c Fix error handling when there are no column families to repair
s f1459540e5 Fix broken nowInSec deserialization
s 91c1befd1c Fix Mutation serializedSize
s b11467c200 Using simulated clock instead of raw nanoTime
s 6412b35924 Accord repair needs to use sentinel tokens
s d19c01ff01 Enable Accord repair with HappyPathFuzzTest
s 2da87f91b8 Implement abort in AccordRepairJob
s e12877c4fa move barrier to accord spec

Patch by Ariel Weisberg; Reviewed by David Capwell for CASSANDRA-19023

krummas committed:
Merge branch 'cassandra-5.0' into trunk

jacek-lewandowski committed:
Fix repeated tests on CircleCI and long-testsome/burn-testsome targets

Patch by Jacek Lewandowski; reviewed by Berenguer Blasi for CASSANDRA-19086

ifesdjeen committed:
Reduce time the tests run to avoid timeouts

Patch by Alex Petrov; reviewed by Marcus Eriksson for CASSANDRA-19123

yifan-c committed:
Merge branch 'cassandra-4.0' into cassandra-4.1

yifan-c committed:
Merge branch 'cassandra-4.1' into cassandra-5.0

yifan-c committed:
Merge branch 'cassandra-5.0' into trunk

ifesdjeen committed:
Remove relocated.shadedad.io.netty.transport.noNative prop

Patch by Alex Petrov, reviewed by Francisco Guerrero for CASSANDRA-19119;

ifesdjeen committed:
Merge branch 'cassandra-5.0' into trunk

ifesdjeen committed:
Remove dependency on bundled Harry jar

Patch by Alex Petrov; reviewed by Marcus Eriksson and Sam Tunnicliffe for CASSANDRA-19083

ifesdjeen committed:
Fix ConsistentMoveTest: nodes were getting stuck in assertGossipStatus, since we would log the message, but never change the actual gossip status.

Patch by Alex Petrov; reviewed by Marcus Eriksson for CASSANDRA-19123

ifesdjeen committed:
Improve the situation with timeouts of all fuzz tests.

Since we would wait for request timeout, even if one progress barrier message gets lost,
we would fail the test, since we would wait for 10 minutes of timeout. But since _commit_
is paused, we could just attempt to retry, and only one of the attempts will succeed.

Patch by Alex Petrov; reviewed by Marcus Eriksson for CASSANDRA-19123

krummas committed:
Test failure: materialized_views_test.TestMaterializedViewsConsistency.test_multi_partition_consistent_reads_after_write

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19056

beobal committed:
Rewrite RegisterTest to verify serialization version ceiling

Patch by Sam Tunnicliffe; reviewed by Marcus Eriksson for CASSANDRA-19073

michaelsembwever committed:
ninja-fix add addtocmstool tool to bin directory in deb/rpm packages

ifesdjeen committed:
Fix test Failure: MixedModeFrom3LoggedBatchTest.testSimpleStrategy

The problem was that read command is deserialising serialisedAt epoch as null by default and not as Empty, causing NPE.

Patch by Alex Petrov; reviewed by Sam Tunnicliffe CASSANDRA-19066.

michaelsembwever committed:
ninja-fix add addtocmstool tool to bin directory in rpm noboolean package

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Fix SAI intersection queries

michaelsembwever committed:
ninja-fix changes for 19011

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  ninja-fix changes for 19011

michaelsembwever committed:
Prepare debian changelog for 5.0-beta1

krummas committed:
Lower the number of tables and repairs to avoid timeouts in ClearSnapshotTest

Patch by marcuse; reviewed by Sam Tunnicliffe for CASSANDRA-19070

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Prepare debian changelog for 5.0-beta1

driftx committed:
Revert "Upgrade jackson to 2.15.3 and snakeyaml to 2.1"

This reverts commit 64ae866095f0371daabbb82ac3cb80e979d79188.

jacek-lewandowski committed:
Merge branch 'cassandra-4.0' into cassandra-4.1

* cassandra-4.0:
  Fix repeated tests on CircleCI and long-testsome/burn-testsome targets

jacek-lewandowski committed:
Merge branch 'cassandra-4.1' into cassandra-5.0

* cassandra-4.1:
  Fix repeated tests on CircleCI and long-testsome/burn-testsome targets

jacek-lewandowski committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Fix repeated tests on CircleCI and long-testsome/burn-testsome targets

driftx committed:
Revert "Revert "Upgrade jackson to 2.15.3 and snakeyaml to 2.1""

This reverts commit 8cd0690c0bfca55808d25b1d24fa06e6d65930e5.

beobal committed:
In in-jvm dtests, don't mark internode messaging as started until it actually is

Patch by Sam Tunnicliffe; reviewed by Marcus Eriksson for CASSANDRA-19087

michaelsembwever committed:
Write ccm clusters under test's TMPDIR

This prevents docker overlay file leakage from containers by keeping write-heavy directories in bind mounts (or volumes). It also offers better performance by avoiding the container filesystem.

 patch by Mick Semb Wever; reviewed Brandon Williams for INFRA-25202

michaelsembwever committed:
Merge branch 'cassandra-5.0' into trunk

* cassandra-5.0:
  Write ccm clusters under test's TMPDIR

aweisberg committed:
Fix AccordCommandsPurger universal durability check

aweisberg committed:
Fix AccordJournal.Type interop entries

Patch by Ariel Weisberg; Reviewed by David Capwell for CASSANDRA-19024

bdeggleston committed:
Fix empty row check in command/tfk mutation methods

bdeggleston committed:
Fix CompactionAccordIteratorsTest

aweisberg committed:
Fix Mutation serializer

aweisberg committed:
Fix 5.1 rebase bugs

aweisberg committed:
Add back enforceInvariants=false to shouldCleanup in AccordCommandsPurger

aweisberg committed:
Fix StorageAttachedIndexDDLTest failing due to background Accord compactions

aweisberg committed:
Instance.receiveMessage should use sync

We mistakenly set this to async thinking async delivery was needed, but whether it is async or not is based on the boolean paramter passed to it.

Right now SchemaTest relies on it running sync, deserializaing in the other class loader and generating an error, and then handling that error from inside C*. This isn't how it actually works on failed deserialize (it's just a timeout), but the test relies on it so for now not changing it.

aweisberg committed:
Fix GuardrailTablesTest

aweisberg committed:
Don't use ImmediateExectuor in JVM dtests to process messages

For now just throwing it on MISC stage which is an unbounded thread pool that shouldn't deadlock

aweisberg committed:
Fix TombstoneCountWarningTest

bdeggleston committed:
Schema based accord fast path configuration

Patch by Blake Eggleston; Reviewed by David Capwell and Alex Petrov for CASSANDRA-19009

dcapwell committed:
(Accord): Bug fixes from CASSANDRA-18675 to better support adding keyspaces

patch by David Capwell; reviewed by Benedict Elliott Smith, Blake Eggleston for CASSANDRA-18804

dcapwell committed:
(Accord) NPE while trying to serialize FoundKnownMap as value is null half the time but unexpected while serializing

patch by David Capwell; reviewed by Blake Eggleston for CASSANDRA-19253

ifesdjeen committed:
Quick fix for AccordAddTableTest: make rejections catch up

bdeggleston committed:
fix fast path coordinator test

dcapwell committed:
Ninja: Added check that command.route is not null when SPL is trying to inform, saw this event in CI and need more details to know what happened

bdeggleston committed:
retry on Invalidated

bdeggleston committed:
fix ReducingRangeMapSerializer

bdeggleston committed:
improve accord error logging

bdeggleston committed:
don't make noise when fast path reconfigs are rejected

bdeggleston committed:
apply cfk update to both cfk caches

bdeggleston committed:
don't mark initialized cfk instances as updated

bdeggleston committed:
fix fast path update deadlock

belliottsmith committed:
CASSANDRA-18365: Protocol fixes

iamaleksey committed:
Fix LocalPartitioner duplication in *_for_key Accord system tables

patch by Aleksey Yeschenko; reviewed by Marcus Eriksson for
CASSANDRA-19265

bdeggleston committed:
load both key command histories for operations specifying keyhistory ALL

bdeggleston committed:
fix fast path test epoch wait

belliottsmith committed:
Fix SerializerSupport.reconstruct, and introduce RangeFactory

belliottsmith committed:
Fast single-partition "Ephemeral Reads"

Introduce a special kind of non-durable read that provides only per-key linearizable isolation; i.e. strict-serializable isolation for single partition-key reads.
This read creates only a happens-before edge, by collecting dependencies for execution and ensuring that execution happens strictly after these dependencies
have executed, but at no precise time otherwise. So later writes may be witnessed, and if multiple keys are read they may represents different points in time.

patch by Benedict; reviewed by Ariel Weisberg for CASSANDRA-19305

belliottsmith committed:
Fix (de)serialization of WaitingOn into cache

belliottsmith committed:
Fix simulator logging and clean shutdown

belliottsmith committed:
dtest Instance shutdown (again)

belliottsmith committed:
Refactor CommandsForKey for efficiency, and to support transitive dependency elision

patch by Benedict; reviewed by Aleksey Yeshchenko for CASSANDRA-19310

aweisberg committed:
Add TODO for TxnQuery using wrong cluster metadata

bdeggleston committed:
CEP-15: (C*) per-table transactional configuration

Patch by Blake Eggleston; Reviewed by Ariel Wesberg for CASSANDRA-19016

aweisberg committed:
Revert "CEP-15: (C*) per-table transactional configuration"

This reverts commit 85411ebcdd94a5b37b668ab16afc69c8443bc42c.
